src/carnot/planner/probes/probes_test.cc

```diff
@@ -648,6 +648,228 @@ TEST_F(ProbeCompilerTest, parse_bpftrace) {
               testing::proto::EqualsProto(absl::Substitute(kBPFTraceProgramPb, literal_bpf_trace)));
 }
 
+constexpr char kBPFTraceProgramMaxKernel[] = R"bpftrace(
+kprobe:tcp_drop
+{
+  ...
+}
+)bpftrace";
+
+constexpr char kBPFTraceProgramMinKernel[] = R"bpftrace(
+tracepoint:skb:kfree_skb
+{
+  ...
+}
+)bpftrace";
+
+// Test that we can compile/parse a single TraceProgram object with a valid selector
+constexpr char kBPFSingleTraceProgramObjectPxl[] = R"pxl(
+import pxtrace
+import px
+
+after_519_trace_program = pxtrace.TraceProgram(
+  program="""$0""",
+  min_kernel='5.19',
+)
+
+table_name = 'tcp_drop_table'
+pxtrace.UpsertTracepoint('tcp_drop_tracer',
+                          table_name,
+                          after_519_trace_program,
+                          pxtrace.kprobe(),
+                          '10m')
+)pxl";
+
+constexpr char kBPFSingleTraceProgramObjectPb[] = R"proto(
+name: "tcp_drop_tracer"
+ttl {
+  seconds: 600
+}
+programs {
+  table_name: "tcp_drop_table"
+  bpftrace {
+    program: "$0"
+  }
+  selectors {
+    selector_type: MIN_KERNEL
+    value: "5.19"
+  }
+}
+)proto";
+
+TEST_F(ProbeCompilerTest, parse_single_bpftrace_program_object) {
+  ASSERT_OK_AND_ASSIGN(auto probe_ir,
+                       CompileProbeScript(absl::Substitute(kBPFSingleTraceProgramObjectPxl,
+                                                           kBPFTraceProgramMinKernel)));
+  plannerpb::CompileMutationsResponse pb;
+  EXPECT_OK(probe_ir->ToProto(&pb));
+  ASSERT_EQ(pb.mutations_size(), 1);
+
+  std::string literal_bpf_trace_min = kBPFTraceProgramMinKernel;
+  literal_bpf_trace_min =
+      std::regex_replace(literal_bpf_trace_min, std::regex(R"(\\\n)"), R"(\\\\n)");
+  literal_bpf_trace_min = std::regex_replace(literal_bpf_trace_min, std::regex("\n"), "\\n");
+  literal_bpf_trace_min = std::regex_replace(literal_bpf_trace_min, std::regex("\""), "\\\"");
```

Comment:
Do we have to use regular expressions? It seems the other tests use `absl::Substitute` and imo it's much more readable.

Reply:
Thanks for pointing that out - I've removed a few calls to `regex_replace` which I believe are unneeded.

I think `absl::Substitute` is used for parameterized string substitution rather than pattern-based substitution which we're doing here. The existing test case for bpftrace strings on lines 635-649 also makes use of `regex_replace`, so I simply copied those over. They were [added in this commit.](https://github.com/benkilimnik/pixie-privy/commit/1593bcb2d31eaff8eb5d8b207d7ad574f0696b17)

Another option would be to use [absl::StrReplaceAll](https://github.com/abseil/abseil-cpp/blob/master/absl/strings/str_replace.h#L194) but I ran into the `error: no member named 'StrReplaceAll' in namespace 'absl'`